### PR TITLE
feat / add argument details and values to single-market strategy classes

### PR DESCRIPTION
**Pull request by kmcgowan2000**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:

### CODE REVIEW

**Comment by phbrgnomo** [comment]
@dennisocana This is the PR that was mentioned on latest product meeting.

@kmcgowan2000 Could you create an Issue linked to this PR describing what is the goal?

**Comment by kmcgowan2000** [comment]
> @dennisocana This is the PR that was mentioned on latest product meeting.
> 
> @kmcgowan2000 Could you create an Issue linked to this PR describing what is the goal?

@phbrgnomo yeah that's created and linked now, and the first commit to this PR is a completed example (PMM strat)

File: setup.py [hunk 1]
```diff
@@ -102,6 +106,11 @@ def main():
         ],
     }
     install_requires = [
+        "0x-contract-addresses",
```
Comment (Nullably) [inline setup.py:109]: Why do we need all of these 0x libraries?

**Review by Nullably — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (kmcgowan2000) [reply setup.py:109]: can't import from hummingbot.connector.utils without them

Comment (kmcgowan2000) [reply setup.py:109]: I guess we could split it off into a specific extras_require ? if it's only really useful for hummingsim users

**Comment by phbrgnomo** [comment]
These new parameters are being created to allow Hbot pro read the information of the strategy configuration.

The PR cover only strategies that trade on a single pair.

After that, we need to expand these values to other strategies, to allow Hbot pro also access this information.

We also need to add on the development docs an explanation about what these parameters are used for. @kmcgowan2000 can help on writing the docs.

@dennisocana PR is ready to review and merge. The test is related to Hbot pro, so after approving the PR, contact @kmcgowan2000 for tests.

File: setup.py (hunk 1 above)
Comment (Nullably) [reply setup.py:109]: I think that's fine, I just couldn't find where they are used.

**Review by Nullably — APPROVED** [review]
LGTM

**Comment by Nullably** [comment]
@phbrgnomo @dennisocana 
Note that not all strategies and connectors can be used by Hummingsim (thus Hummingbot Pro). Python strategies will need to first be converted to Cython. Gateway and derivative connectors are not available as we never collect order book data for them (or they don't have order books).

**Comment by Nullably** [comment]
A simpler solution is proposed. I'm closing this one.

**Pull request closed without merging**